packages/amplify-category-geo/src/provider-controllers/geofenceCollection.ts

```diff
@@ -0,0 +1,76 @@
+import { printNextStepsSuccessMessage, setProviderContext, insufficientInfoForUpdateError } from './index';
+import { ServiceName } from '../service-utils/constants';
+import { printer } from 'amplify-prompts';
+
+export const addGeofenceCollectionResource = async (
+  context: $TSContext
+): Promise<string> => {
+  // initialize the Geofence Collection parameters
+  let geofenceCollectionParams: Partial<GeofenceCollectionParameters> = {
+    providerContext: setProviderContext(context, ServiceName.GeofenceCollection)
+  };
+  // populate the parameters for the resource
+  await createGeofenceCollectionWalkthrough(context, geofenceCollectionParams);
+  const completeParameters: GeofenceCollectionParameters = convertToCompleteGeofenceCollectionParams(geofenceCollectionParams);
+
+  await createGeofenceCollectionResource(context, completeParameters);
+
+  printer.success(`Successfully added resource ${completeParameters.name} locally.`);
+  printNextStepsSuccessMessage(context);
+  return completeParameters.name;
+};
+
+export const updateGeofenceCollectionResource = async (
+  context: $TSContext
+): Promise<string> => {
+  let geofenceCollectionParams: Partial<GeofenceCollectionParameters> = {
+    providerContext: setProviderContext(context, ServiceName.GeofenceCollection)
+  };
+  // populate the parameters for the resource
+  await updateGeofenceCollectionWalkthrough(context, geofenceCollectionParams);
+  const completeParameters: GeofenceCollectionParameters = convertToCompleteGeofenceCollectionParams(geofenceCollectionParams);
+
+  await modifyGeofenceCollectionResource(context, completeParameters);
+
+  printer.success(`Successfully updated resource ${geofenceCollectionParams.name} locally.`);
+  printNextStepsSuccessMessage(context);
+  return completeParameters.name;
+};
+
+export const removeGeofenceCollectionResource = async (
+  context: any
+): Promise<string | undefined> => {
+  const { amplify } = context;
+  const resourceToRemove = await removeWalkthrough(context, ServiceName.GeofenceCollection);
+  if (!resourceToRemove) return;
+
+  const resourceParameters = await getCurrentGeofenceCollectionParameters(resourceToRemove);
+
+  try {
+    await amplify.removeResource(context, category, resourceToRemove)
+    .then(async (resource: { service: string; resourceName: string }) => {
+      if (resource?.service === ServiceName.GeofenceCollection && resourceParameters.isDefault) {
+        // choose another default if removing a default geofence collection
+        await updateDefaultGeofenceCollectionWalkthrough(context, resource.resourceName);
+      }
+    });
+  } catch (err: $TSAny) {
+    if (err.stack) {
+      printer.error(err.stack);
+      printer.error(err.message);
+      printer.error(`An error occurred when removing the geo resource ${resourceToRemove}`);
+    }
+
+    context.usageData.emitError(err);
+    process.exitCode = 1;
+  }
+
+  printNextStepsSuccessMessage(context);
```

Comment:
This will print a message that reminds to `push`/`publish` local changes. 
```
Next steps:
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud
```